src/Core/App.xaml.cs

```diff
@@ -331,18 +331,27 @@ void NavigateToSetPasswordPageAction()
                     || message.Command == "unlocked"
                     || message.Command == AccountsManagerMessageCommands.ACCOUNT_SWITCH_COMPLETED)
                 {
-                    if (message.Command == AccountsManagerMessageCommands.ACCOUNT_SWITCH_COMPLETED)
+                    var userVerificationMediatorService = ServiceContainer.Resolve<IFido2MakeCredentialConfirmationUserInterface>();
+                    if (message.Command == AccountsManagerMessageCommands.ACCOUNT_SWITCH_COMPLETED && userVerificationMediatorService.IsConfirmingNewCredential)
                     {
-                        var userVerificationMediatorService = ServiceContainer.Resolve<IFido2MakeCredentialConfirmationUserInterface>();
                         userVerificationMediatorService?.OnConfirmationException(new AccountSwitchedException());
                     }
-                    
+
                     lock (_processingLoginRequestLock)
                     {
                         // lock doesn't allow for async execution
                         CheckPasswordlessLoginRequestsAsync().Wait();
                     }
                 }
+                else if (message.Command == "navigateTo" && message.Data is NavigationTarget navigationTarget)
```

Comment:
⛏️ If possible, move `navigateTo` into a constant somewhere to be reused when the message is sent and here in the receiver.